File: branches/cpp-impl/src/widgets/DescriptionEdit.cpp
/* 
 * File:   DescriptionEdit.cpp
 * Author: jolo
 * 
 * Created on 10. Dezember 2009, 20:54
 */

#include "DescriptionEdit.h"

#include <QLabel>

DescriptionEdit::DescriptionEdit(QWidget* p)
: QTextEdit(p),
_lbl(0) {
    connect(this, SIGNAL(textChanged()), this, SLOT(slotTextChanged()));
}

DescriptionEdit::~DescriptionEdit() {
}

QLabel* DescriptionEdit::getLabel() {
    if (!_lbl) {
        _lbl = new QLabel(tr("&Description"), (QWidget*)parent());
        _lbl->setBuddy(this);
        _lbl->setAlignment(Qt::AlignTop | Qt::AlignRight);
    }

    return _lbl;
}

void DescriptionEdit::slotTextChanged() {
    emit descriptionChanged(document()->toPlainText());
}

File: branches/python-impl/src/Gui/Widgets/DescriptionEdit.py
from PyQt4.QtGui import *
from PyQt4.QtCore import *

class DescriptionEdit(QTextEdit):
    descriptionChanged = pyqtSignal(QString)

    def __init__(self, parent):
        QLineEdit.__init__(self, parent)
        self.textChanged.connect(self.onTextChanged)
    def onTextChanged(self):
        self.descriptionChanged.emit(unicode(self.toPlainText()))
    def setValue(self, txt):
        self.setPlainText(unicode(txt))
